crates/ra_proc_macro/src/msg.rs

```diff
@@ -0,0 +1,218 @@
+//! A simplified version of lsp base protocol for rpc
```

Comment:
I feel like we can further strip this down:

* the protocol can be assymentric. We have only requests flowing in one direction and responses in other direction
* the protocol can be blocking: we wait for each response, so we don't need request IDs
* we can hard-code the set of messages as `enum Requst {ListMacros {...}, CustomDerive{...}}` instead of being generic over `T: Serizelize`
* for wire format, we should just json-per ([ndjson](http://ndjson.org/)). buf.read_line().and_then(serde_json::from_str)

Reply:
Fixed.